### PR TITLE
Fix private registry testcases for package plugin integration test

### LINKED ISSUE #1141: Fix secret registry plugin integration tests for tanzu framework
Some of the package plugin integration tests are disabled. Need to fix the testcases and bring them up.
https://github.com/vmware-tanzu/tanzu-framework/blob/d05775ced0163170f84f45206414a7be2f344eff/pkg/v1/tkg/test/tkgpackageclient/package_plugin_integration_test.go#L388-L440

### PR DESCRIPTION
**Pull request by blc1996**
Signed-off-by: Lucheng Bao <luchengb@vmware.com>

### What this PR does / why we need it

Use a in cluster private registry for package plugin authentication test

### Which issue(s) this PR fixes

Fixes #1141 

### Release note

```release-note
None
```

### PR Checklist

- [x] Squash the commits into one or a small number of logical commits
      
- [x] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [x] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

**Review by shivaani0505 — APPROVED** [review]
LGTM

**Comment by blc1996** [comment]
Need to fix #1274 because the service account password contains whitespace.

File: pkg/v1/tkg/test/tkgpackageclient/package_plugin_integration_test.go [hunk 1]
```diff
@@ -384,60 +397,178 @@ func cleanup() {
 	Expect(resultImgPullSecret.Error).ToNot(HaveOccurred())
 }
 
+func setUpPrivateRegistry(kubeconfigPath, clusterName string) {
+	By("set up private registry in cluster")
+	kubeCtx := clusterName + "-admin@" + clusterName
```
Comment (maralavi) [inline pkg/v1/tkg/test/tkgpackageclient/package_plugin_integration_test.go:402]: Do we need to add clusterName as config value in test-infra tests in that case? Cause right now it is not provided there, cause cluster name was not mandatory till now.
https://gitlab.eng.vmware.com/TKG/test-infra/-/blob/master/tests/e2e/byot/extensions/test_14_tkg_package_plugin.py#L53

Comment (blc1996) [reply pkg/v1/tkg/test/tkgpackageclient/package_plugin_integration_test.go:402]: Actually, I think I can remove the cluster name here. As I assume the default context would point to the correct cluster for those kubeconfig paths?

Comment (maralavi) [reply pkg/v1/tkg/test/tkgpackageclient/package_plugin_integration_test.go:402]: I think so, but not sure if you need to hardcode kubecontext for Corgi management and workload clusters. Can check once the PR was merged and the tests ran or running tests locally as per [these instructions](https://gitlab.eng.vmware.com/TKG/test-infra/-/tree/master/tests) and see if they work as expected.

How about local integration tests. I mean does running ```ginkgo -tags embedproviders pkg/v1/tkg/test/tkgpackageclient/``` for a hardcoded cluster kubeconfig (in order to not create a cluster from scratch) work now?

Comment (blc1996) [reply pkg/v1/tkg/test/tkgpackageclient/package_plugin_integration_test.go:402]: After second thought, we do need cluster name provided for local testing.

**Review by vijaykatam — COMMENTED** [review]
Change looks good. Will approve once checks pass.

File: pkg/v1/tkg/test/tkgpackageclient/config/assets/instructions-for-one-time-setup.txt [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+# grab the files from the mountpoint we made and stored them in a configmap
+## tar them
+tar -cf tmp-registry -P /tmp/registry
+## make the configmap
+kubectl create configmap registry-contents --from-file=./tmp-registry
+kubectl get configmap registry-contents -o yaml > registry-contents.yml
+## clean up temporary artifacts
+rm tmp-registry                                                                                                                                                                                   [16/08/21| 5:12PM]
+kubectl delete configmap registry-contents
```
Comment (maralavi) [inline pkg/v1/tkg/test/tkgpackageclient/config/assets/instructions-for-one-time-setup.txt:16]: This ```[16/08/21| 5:12PM]``` is redundant

Comment (blc1996) [reply pkg/v1/tkg/test/tkgpackageclient/config/assets/instructions-for-one-time-setup.txt:16]: Removed

**Review by maralavi — APPROVED** [review]

File: pkg/v1/tkg/test/tkgpackageclient/package_plugin_integration_test.go (hunk 1 above)
Comment (blc1996) [reply pkg/v1/tkg/test/tkgpackageclient/package_plugin_integration_test.go:402]: Discussed offline with @maralavi , we need clusterName for local testing

**Review by vijaykatam — APPROVED** [review]

**Pull request merged**